cli/parser/parsed/parsed.go

```diff
@@ -614,9 +615,17 @@ func (a *OrderedArgs) ConsumeRCFileOptions(workspaceDir string) (rcFiles []strin
 				rcFiles = append(rcFiles, filepath.Join(workspaceDir, ".bazelrc"))
 			}
 		case "home_rc":
-			usr, err := user.Current()
-			if err == nil {
-				rcFiles = append(rcFiles, filepath.Join(usr.HomeDir, ".bazelrc"))
+			// Use $HOME or %USERPROFILE% to locate the home_rc file.
+			// This enables mocking $HOME in test environments.
+			//
+			// On Unix, if $HOME variable is unset, fallback to finding home directory
+			// by syscall (getpwuid), which typically parse /etc/passwd for the information.
+			if homeDir, osErr := os.UserHomeDir(); osErr != nil && homeDir != "" {
```

Comment:
Any chance of this change breaking anyone?

Reply:
I expect this to be more consistent with how Bazel is looking for home_rc file.

https://cs.opensource.google/bazel/bazel/+/master:src/main/cpp/blaze_util_posix.cc;l=229;drc=f50146edd2c8825346dc81650beacd2db0d67afa